source/common/http/http3/conn_pool.cc

```diff
@@ -109,7 +109,11 @@ void Http3ConnPoolImpl::onConnectFailed(Envoy::ConnectionPool::ActiveClient& cli
 }
 
 // Make sure all connections are torn down before quic_info_ is deleted.
-Http3ConnPoolImpl::~Http3ConnPoolImpl() { destructAllConnections(); }
+Http3ConnPoolImpl::~Http3ConnPoolImpl() {
+  destructAllConnections();
+  // set this to zero since state_ is not tracking connections for H3, only the initial capacity
+  state_.connecting_and_connected_stream_capacity_ = 0;
```

Comment:
Is this the only piece of state we should consider resetting? Or maybe just ASSERT() that pending_streams_ and active_streams are zero?

Reply:
you're right, just removing the assertion on connecting_and_connected_stream_capacity_ and keeping pending_streams_ and active_streams makes more sense